model.py

```diff
@@ -11402,11 +11418,12 @@ def site_configuration_has_changed(_db, timeout=1):
 
         # Update the timestamp.
         now = datetime.datetime.utcnow()
-        sql = "UPDATE timestamps SET timestamp=:timestamp WHERE service=:service AND collection_id IS NULL;"
+        earlier = now-datetime.timedelta(seconds=1)
+        sql = "UPDATE timestamps SET timestamp=:timestamp WHERE service=:service AND collection_id IS NULL AND timestamp<=:earlier;"
```

Comment:
Adding this clause to the SQL statement makes it likely that if two different processes run this code nearly simultaneously, the second one will end without doing anything because its UPDATE statement will match no rows.